collection-scripts/gather_monitoring

```diff
@@ -1,29 +1,208 @@
+
+declare -r BASE_COLLECTION_PATH="/must-gather"
+declare -r MONITORING_PATH="${BASE_COLLECTION_PATH}/monitoring"
+declare -r PROM_PATH="${MONITORING_PATH}/prometheus"
+declare -r CA_BUNDLE="${MONITORING_PATH}/ca-bundle.crt"
+
+# Setting: Session token, overwriten by GATHER_MONIT_TOKEN
+declare -r SS_TOKEN="${GATHER_MONIT_TOKEN:-$(oc whoami -t)}"
+declare -r PROM_HOST=$(oc -n "${MONIT_NS}" get route prometheus-k8s -o jsonpath='{.spec.host}{"\n"}')
+declare -r PROM_API="/api/v1"
+
+function get_env {
+    oc -n "${MONIT_NS}" extract "cm/${MONIT_ENV}" \
+        --keys=env \
+        --to="${MONITORING_PATH}/" 2>&1 >/dev/null || true
+
+    echo "INFO: Get custom metrics from ConfigMap ${MONIT_ENV} on project ${MONIT_NS}"
+    if [[ -s "${MONITORING_PATH}/env" ]]; then
+        echo "INFO: Loading custom environments variables from ${MONITORING_PATH}/env"
+        source "${MONITORING_PATH}/env"
+    else
+        echo "INFO: Unable to load custom environments from ConfigMap, ignoring."
+    fi
+}
+
+# this is a CA bundle we need to verify the monitoring route,
+# we will write it to disk so we can use it in the flag
+function get_ca_bundle {
+    oc -n openshift-config-managed \
+        get cm default-ingress-cert \
+        -o jsonpath='{.data.ca-bundle\.crt}' > "${CA_BUNDLE}"
+}
+
+function init {
+
+    mkdir -p "${PROM_PATH}"
+
+    get_ca_bundle
+
+    get_env
+
+    # Query Param: 'start' and 'end' timestamp
+    local _START_HUMAN=${GATHER_MONIT_START_DATE:-${GATHER_MONIT_START_DATE_DEFAULT}}
+    local _END_HUMAN=${GATHER_MONIT_END_DATE:-${GATHER_MONIT_END_DATE_DEFAULT}}
+    declare -g DATE_START=$(date -d "${_START_HUMAN}" +%s)
+    declare -g DATE_END=$(date -d "${_END_HUMAN}" +%s)
+
+    # Query Param: 'step'. Low metric resolution with long range could be limited to 11k datapoints.
+    declare -g QUERY_STEP=${GATHER_MONIT_QUERY_STEP:-${GATHER_MONIT_QUERY_STEP_DEFAULT}}
+
+    declare -g PROM_API_QSTR="start=${DATE_START}&end=${DATE_END}&step=${QUERY_STEP}"
+
+    cat <<-EOF
+INFO: Metrics config time range from=${DATE_START} to=${DATE_END} step=${QUERY_STEP}
+INFO: Metrics human  time range from=[$(date -d "@${DATE_START}")] to [$(date -d "@${DATE_END}")]
+INFO: Config Env GATHER_MONIT_START_DATE: ${GATHER_MONIT_START_DATE:-""}
+INFO: Config Env GATHER_MONIT_END_DATE: ${GATHER_MONIT_END_DATE:-""}
+INFO: Config Env GATHER_MONIT_QUERY_STEP: ${GATHER_MONIT_QUERY_STEP:-""}
+INFO: Config Env GATHER_MONIT_CUSTOM_METRICS: ${GATHER_MONIT_CUSTOM_METRICS:-""}
+INFO: Config Env GATHER_MONIT_DISCOVERY_PREFIXES: ${GATHER_MONIT_DISCOVERY_PREFIXES:-""}
+INFO: Prometheus endpoint: https://${PROM_HOST}
+INFO: Prometheus base query: ${PROM_API}/<uri>?query=<metric>&${PROM_API_QSTR}"
+EOF
+}
+
+function cleanup {
+    rm "${CA_BUNDLE}"
+}
+
+function oc_get {
+    oc get \
+        --certificate-authority="${CA_BUNDLE}" \
+        --token="${SS_TOKEN}" \
+        "$@"
+}
+
+function prom_get_api {
+    local API_URI="${PROM_API}/${1}"; shift
```

Comment:
NIT: could you please use UPPER for `globals` and snake_case for `locals` ?

Reply:
Done https://github.com/mtulio/must-gather/pull/21/commits/e6b6b276402ea92942571ce5461b03266dad8bfb